v23.1/connect-to-a-changefeed-kafka-sink-with-oauth-using-okta.md

```diff
@@ -0,0 +1,291 @@
+    ~~~ shell
+    mvn package
+    ~~~
+
+## Step 5. Add OAuth environment variables
+
+In this step, you'll export the environment variables that contain your OAuth client credentials. You will need the following information from your Okta developer account:
+
+- Your Okta developer account URL, for example `dev-12345678.okta.com`
+- The absolute paths to your: 
+    - `kafka-oauth` directory
+    - `kafka_2.13-2.8.2` directory (or your Kafka version)
+- Your client ID and client secret from your Okta application (created in [Step 1](#step-1-create-an-okta-application-and-scope)). You will need to build the value for the variable as `{client ID}:{client secret}` and [base64 encode](https://www.base64encode.org/) the whole value.
+
+Export the following environment variables, or add them to your configuration file (e.g., `.bash_profile` or `.bashrc`) replacing the values between `{..}` with those mentioned in the preceding list:
+
+~~~
+export OAUTH_WITH_SSL=true
+export OAUTH_LOGIN_SERVER="{Okta developer account URL}"
+export OAUTH_LOGIN_ENDPOINT="/oauth2/default/v1/token"
+export OAUTH_LOGIN_GRANT_TYPE="client_credentials"
+export OAUTH_LOGIN_SCOPE="kafka"
+export OAUTH_AUTHORIZATION="Basic {base64-encoded client credential value}"
+export OAUTH_INTROSPECT_SERVER="{Okta developer account URL}"
+export OAUTH_INTROSPECT_ENDPOINT="/oauth2/default/v1/introspect"
+export OAUTH_INTROSPECT_AUTHORIZATION="Basic {base64-encoded client credential value}"
+
+export CLASSPATH="{absolute path to kafka-oauth}/target/*"
+export KAFKA_OPTS="-Djava.security.auth.login.config={absolute path to kafka_2.13-2.8.2}/config/kafka_server_jaas.conf"
+~~~
+
+## Step 6. Start your Kafka server
+
+Ensure you're in the `kafka_2.13-2.8.2` directory, and then run the following:
+
+1. Start the Zookeeper server in one terminal:
+
+    {% include_cached copy-clipboard.html %}
+    ~~~ shell
+    ./bin/zookeeper-server-start.sh ./config/zookeeper.properties
+    ~~~
+
+1. Start the Kafka server in another terminal:
+
+    {% include_cached copy-clipboard.html %}
+    ~~~ shell
+    ./bin/kafka-server-start.sh ./config/server.properties
+    ~~~
+
+1. In another separate terminal window, create a topic to consume:
+
+    {% include_cached copy-clipboard.html %}
+    ~~~ shell
+    ./bin/kafka-topics.sh --bootstrap-server localhost:9092 --create --topic vehicles
+    ~~~
+
+    To consume messages for this topic, run:
+
+    {% include_cached copy-clipboard.html %}
+    ~~~ shell
+    ./bin/kafka-console-consumer.sh --bootstrap-server localhost:9092 --topic vehicles
+    ~~~
+
+## Step 7. Create a changefeed
+
+In this step, you will create a changefeed authenticating with Okta.
+
+The Kafka URI must follow this format:
+
+~~~
+'kafka://{kafka cluster address}:9093?topic_name={vehicles}&sasl_client_id={your client ID}&sasl_client_secret={your base64-encoded client secret}&sasl_enabled=true&sasl_mechanism=OAUTHBEARER&sasl_token_url={your url-encoded Okta developer token URL}'
```

Comment:
```suggestion
kafka://{kafka cluster address}:9093?topic_name={vehicles}&sasl_client_id={your client ID}&sasl_client_secret={your base64-encoded client secret}&sasl_enabled=true&sasl_mechanism=OAUTHBEARER&sasl_token_url={your url-encoded Okta developer token URL}
```

Reply:
I'm actually going to add a copy-clipboard to this and keep the string quotes so that users can easily lift this and replace what they need to.

Reply:
Makes sense - thanks.